### PR TITLE
Creating Cucumber Tests for shipping notification (Fact Acct)

### PR DESCRIPTION
**Pull request by mi-ps**
metasfresh issue : https://github.com/metasfresh/metasfresh/issues/16337

### CODE REVIEW

File: backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/shippingNotification/shippingNotification.feature [hunk 1]
```diff
@@ -51,8 +51,19 @@ Feature: Shipping Notifications
     And metasfresh contains C_Auction:
       | Identifier | Name     | Date       |
       | Auction_1  | Auction1 | 2023-09-21 |
+    And load C_Element:
+      | C_Element_ID.Identifier | OPT.C_Element_ID |
+      | element_1               | 1000000          |
+    And load C_ElementValue:
+      | C_ElementValue_ID.Identifier | C_Element_ID.Identifier | Value |
+      | elementValue_1               | element_1               | 90000 |
```
Comment (teosarca) [inline backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/shippingNotification/shippingNotification.feature:59]: use P_Asset_Acct as the identifier

File: backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/shippingNotification/shippingNotification.feature [hunk 2]
```diff
@@ -51,8 +51,19 @@ Feature: Shipping Notifications
...
+      | elementValue_2               | element_1               | 90055 |
```
Comment (teosarca) [inline backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/shippingNotification/shippingNotification.feature:60]: use P_ExternallyOwnedStock_Acct as the identifier

(i.e. something which is human/tester friendly, easy to read and understand)

File: backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/accountingShippingNotification.feature [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID.Identifier | DateOrdered | OPT.C_PaymentTerm_ID | OPT.POReference | OPT.DocBaseType | OPT.M_PricingSystem_ID.Identifier |
+      | o_1        | N       | bpartner_1               | 2021-04-16  | 1000012              | po_ref_mock     | POO             | ps_1                              |
+    And metasfresh contains C_OrderLines:
+      | Identifier | C_Order_ID.Identifier | M_Product_ID.Identifier | QtyEntered |
+      | ol_1       | o_1                   | p_1                     | 20         |
+    When the order identified by o_1 is completed
+    And after not more than 30s, M_ReceiptSchedule are found:
+      | M_ReceiptSchedule_ID.Identifier | C_Order_ID.Identifier | C_OrderLine_ID.Identifier | C_BPartner_ID.Identifier | C_BPartner_Location_ID.Identifier | M_Product_ID.Identifier | QtyOrdered | M_Warehouse_ID.Identifier |
+      | receiptSchedule_04022023_1      | o_1                   | ol_1                      | bpartner_1               | bpLocationDefault                 | p_1                     | 20         | warehouseStd              |
```
Comment (cristinamghita) [inline backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/accountingShippingNotification.feature:63]: Why do we need this purchase part here?

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java [hunk 4]
```diff
@@ -620,6 +621,22 @@ public void shipment_action_expecting_error(@NonNull final String model_UNUSED,
...
+			final I_M_InOut reversalInOut = StepDefUtil.tryAndWaitForItem(timeoutSec, 500, () -> load_reversal_InOut(row));
+
+			final SoftAssertions softly = new SoftAssertions();
+			softly.assertThat(reversalInOut).isNotNull();
+			softly.assertAll();
+
+			final String reversalInOutIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_M_InOut_ID + "." + TABLECOLUMN_IDENTIFIER);
+			shipmentTable.putOrReplace(reversalInOutIdentifier, reversalInOut);
+		}
+	}
```
Comment (cristinamghita) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java:638]: Why not calling softly.assertAll() out of loop?
In the that's the purpose of using soft assertions.

**Review by cristinamghita — CHANGES_REQUESTED** [review]
Check my comments

File: backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/accountingShippingNotification.feature (hunk 3 above)
Comment (mi-ps) [reply backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/accountingShippingNotification.feature:63]: @cristinamghita 

image

**Review by cristinamghita — APPROVED** [review]
lgtm
But,pls wait with merge until build is green

**Pull request merged**